closed/src/java.base/share/classes/openj9/internal/criu/NativePRNG.java

```diff
@@ -245,5 +280,92 @@ public OutputStream run() {
+                if (time - lastRead < MAX_BUFFER_TIME) {
+                    return;
+                } else {
+                    // byte is old, so subtract from counter to shrink buffer
+                    change_buffer--;
+                }
+            } else {
+                // No bytes available, so add to count to increase buffer
+                change_buffer++;
+            }
+
+            // If counter has it a limit, increase or decrease size
+            if (change_buffer > REQ_LIMIT_INC) {
+                new_buffer_size = nextBuffer.length * 2;
+            } else if (change_buffer < REQ_LIMIT_DEC) {
+                new_buffer_size = nextBuffer.length / 2;
+            }
+
+            // If buffer size is to be changed, replace nextBuffer.
+            if (new_buffer_size > 0) {
+                if (new_buffer_size <= MAX_BUFFER_SIZE &&
+                        new_buffer_size >= MIN_BUFFER_SIZE) {
+                    nextBuffer = new byte[new_buffer_size];
+                    if (debug != null) {
+                        debug.println("Buffer size changed to " +
+                                new_buffer_size);
+                    }
+                } else {
+                    if (debug != null) {
+                        debug.println("Buffer reached limit: " +
+                                nextBuffer.length);
+                    }
+                }
+                change_buffer = 0;
+            }
+
+            // Load fresh random bytes into nextBuffer
+            lastRead = time;
+            readFully(nextIn, nextBuffer);
+            buffered = nextBuffer.length;
+        }
+
+        private void implNextBytes(byte[] data) {
+                try {
+                    int data_len = data.length;
+                    int ofs = 0;
+                    int len;
+                    int buf_pos;
+
+                    while (data_len > 0) {
+                        synchronized (LOCK_GET_BYTES) {
+                            ensureBufferValid();
+                            buf_pos = nextBuffer.length - buffered;
+                            if (data_len > buffered) {
+                                len = buffered;
+                                buffered = 0;
+                            } else {
+                                len = data_len;
+                                buffered -= len;
+                            }
+
+                            System.arraycopy(nextBuffer, buf_pos, data, ofs, len);
+                        }
+
+                        ofs += len;
+                        data_len -= len;
+                    }
+                } catch (IOException e){
+                    throw new ProviderException("nextBytes() failed", e);
+                }
+        }
+
+	private void clearRNGBuffers() {
+            if (this.nextBuffer != null) {
+                Arrays.fill(this.nextBuffer, (byte) 0x00);
+                this.nextBuffer = null;
+            }
+	}
```

Comment:
Please indent consistently and only with spaces.

Reply:
updated